extensions/amp-lightbox/1.0/lightbox.js

```diff
@@ -0,0 +1,194 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import {ContainWrapper} from '../../../src/preact/component';
+import {forwardRef} from '../../../src/preact/compat';
+import {setStyle} from '../../../src/style';
+import {
+  useEffect,
+  useImperativeHandle,
+  useRef,
+  useState,
+} from '../../../src/preact';
+import {useStyles} from './lightbox.jss';
+
+const ANIMATION_DURATION = 200;
+const ANIMATION_PRESETS = {
+  'fade-in': [
+    {opacity: 0, visibility: 'visible'},
+    {opacity: 1, visibility: 'visible'},
+  ],
+  'fly-in-top': [
+    {opacity: 0, transform: 'translate(0,-100%)', visibility: 'visible'},
+    {opacity: 1, transform: 'translate(0, 0)', visibility: 'visible'},
+  ],
+  'fly-in-bottom': [
+    {opacity: 0, transform: 'translate(0, 100%)', visibility: 'visible'},
+    {opacity: 1, transform: 'translate(0, 0)', visibility: 'visible'},
+  ],
+};
+
+/**
+ * @param {} value
+ * @return {string}
+ */
+function useValueRef(value) {
+  const ref = useRef();
+  ref.current = value;
+  return ref;
+}
+
+/**
+ * @param {!LightboxDef.Props} props
+ * @param {{current: (!LightboxDef.LightboxApi|null)}} ref
+ * @return {PreactDef.Renderable}
+ */
+function LightboxWithRef(
+  {
+    id,
+    animateIn = 'fade-in',
+    closeButtonAriaLabel,
+    children,
+    initialOpen,
+    onBeforeOpen,
+    onAfterClose,
+    ...rest
+  },
+  ref
+) {
+  // There are two phases to open and close.
+  // To open, we mount and render the contents (invisible), then animate the display (visible).
+  // To close, it's the reverse.
+  // `mounted` mounts the component. `visible` plays the animation.
+  const [mounted, setMounted] = useState(initialOpen);
+  const [visible, setVisible] = useState(false);
+  const classes = useStyles();
+  const lightboxRef = useRef();
+
+  // We are using refs here to refer to common strings, objects, and functions used.
+  // This is because they are needed within `useEffect` calls below (but are not depended for triggering)
+  const animateInRef = useValueRef(animateIn);
+  const onBeforeOpenRef = useValueRef(onBeforeOpen);
+  const onAfterCloseRef = useValueRef(onAfterClose);
+
+  useImperativeHandle(
+    ref,
+    () => ({
+      open: () => {
+        if (onBeforeOpenRef.current) {
+          onBeforeOpenRef.current();
+        }
+        setMounted(true);
+        setVisible(true);
+      },
+      close: () => {
+        setVisible(false);
+      },
+    }),
+    [onBeforeOpenRef]
+  );
+
+  useEffect(() => {
+    const element = lightboxRef.current;
+    if (element == undefined) {
+      return;
+    }
+    let animation;
+    if (visible) {
+      const postVisibleAnim = () => {
+        setStyle(element, 'opacity', 1);
+        setStyle(element, 'visibility', 1);
```

Comment:
Numeric visibility is not a thing, I don't think. Should be `hidden` or `visible`